### PR TITLE
Implemented: Schedulers page

### PR DESCRIPTION
**Pull request by lokeshkose**

### CODE REVIEW

File: src/pages/Home/Home.tsx [hunk 1]
```diff
@@ -31,6 +31,7 @@ export const HomePage: React.FC = () => {
           primaryBtnText={t('logoutBtnLabel')}
           primaryButtonIcon={<LogoutOutlined />}
           primartBtnAction={logoutHandler}
+          primaryBtnId={'logout'}
```
Comment (LalitKushwah) [inline src/pages/Home/Home.tsx]: Create a constant then use it instead of hardcoded string

File: src/pages/Schedulers/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+						danger={true}
+						disabled={!isOrderIdNotEmpty}
+						buttontext={t('schedulers.modalOkText')}
+						onConfirm={onConfirm}
+						loading={isLoading}
+						okText={t('schedulers.popconfirmOkText')}
+						id='confirm-order'
+						cancelText={t('schedulers.popconfirmCancelText')}></Components.Popconfirm>,
+				]}>
+				<SchedulersDeleteOrderForm form={form} onChange={onChangeFormValue}></SchedulersDeleteOrderForm>
```
Comment (LalitKushwah) [inline src/pages/Schedulers/Modal.tsx]: Delete order component should be permission protected
Please create an HOC which accepts componnet and its prop and also the parameter. HOC will decide whether to show/disable the component. Please read more about RBAC

File: src/pages/Schedulers/Schedulers.css [hunk 3]
```diff
@@ -0,0 +1,11 @@
+.Schedulers-Container{
+  margin: 10px;
+}
+.Scheduler-button{
+  min-width: 300px;
+  max-width: 300px !important;
```
Comment (LalitKushwah) [inline src/pages/Schedulers/Schedulers.css]: why we are using important here?

File: src/pages/Schedulers/Schedulers.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+		form.resetFields();
+		setIsModalOpen(true);
+	};
+
+	const handleCancel = () => {
+		setIsModalOpen(false);
+	};
+
+	return (
+		<Layout style={{ minHeight: '100vh' }}>
```
Comment (LalitKushwah) [inline src/pages/Schedulers/Schedulers.tsx]: Please move the inline styling to css files

File: src/pages/Schedulers/Schedulers.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+			<NavBar activeKey={'8'} />
+			<Layout>
+				<PageHeader title='Schedulers'></PageHeader>
```
Comment (LalitKushwah) [inline src/pages/Schedulers/Schedulers.tsx]: Harcoded string should come from translations

File: src/pages/Schedulers/Tree.tsx [hunk 6]
```diff
@@ -0,0 +1,470 @@
...
+	};
+	const LeftSideTreeData: TreeDataNode[] = [
+		{
+			title: t('schedulers.products'),
+			key: '1',
+			icon: <StarOutlined />,
+			children: [
+				{
+					title: (
+						<Components.Popconfirm
```
Comment (LalitKushwah) [inline src/pages/Schedulers/Tree.tsx]: Here all items are sharing the same properties we can create a list in util/constant file and read this and pass all common properties

File: src/pages/Schedulers/Schedulers.css (hunk 3 above)
Comment (lokeshkose) [reply src/pages/Schedulers/Schedulers.css]: No need removed.

File: src/pages/Schedulers/Schedulers.tsx (hunk 4 above)
Comment (lokeshkose) [reply src/pages/Schedulers/Schedulers.tsx]: done

File: src/pages/Schedulers/Schedulers.tsx (hunk 5 above)
Comment (lokeshkose) [reply src/pages/Schedulers/Schedulers.tsx]: done

File: src/pages/Schedulers/Tree.tsx (hunk 6 above)
Comment (lokeshkose) [reply src/pages/Schedulers/Tree.tsx]: Done

File: src/services/Api.ts
Comment (LalitKushwah) [inline src/services/Api.ts:1]: Please fix the indentations

File: src/translations/resources.json
Comment (LalitKushwah) [inline src/translations/resources.json:1]: Please fix conflicts I can see some >>>>>> Heads

File: src/utils/common.ts [hunk 7]
```diff
@@ -1,3 +1,56 @@
...
+// Define enum for CRUD operations
+export enum CrudOperation {
+  CREATE = 'create',
+  READ = 'read',
+  UPDATE = 'update',
+  DELETE = 'delete',
+}
+
+// Define CRUD permissions for user roles
+export const crudPermissions: any = {
```
Comment (LalitKushwah) [inline src/utils/common.ts]: use any as less as possible

File: src/services/Schedulers.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+	UPDATE_BANKS: 'erp/create/bank',
+	UPDATE_MILK_COLLECTION_IN_ERP: 'erp/update/milk/collection/erp',
+};
+
+export const triggerScheduler = async (identifier: string, method: string) => {
+	const END_POINT = SCHEDULERS_API[identifier];
+	try {
+		const repsonse = await Api(END_POINT, {
+			method: method,
+			data: {},
```
Comment (LalitKushwah) [inline src/services/Schedulers.ts]: Why are we sending empty data?

File: src/pages/Schedulers/Schedulers.css [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+}
+.Scheduler-button {
+  min-width: 300px;
+  max-width: 300px;
+}
+.ant-tree-node-content-wrapper {
+  font-size: 16px;
+}
+
+.h-100 {
```
Comment (LalitKushwah) [inline src/pages/Schedulers/Schedulers.css]: Please rename this classname to something meaningful like
`.SchedulersPage-Parent-Container`

File: src/common/Permission/RoutePermission.tsx [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+import { getLoggedInUser } from '../../utils/common';
+
+const RoutePermissionwithRBAC =
+  (allowedRoles: string[]) =>
+  <P extends object>(WrappedComponent: React.ComponentType<P>) => {
+    const RBACComponent: React.FC<P> = (props) => {
+      const userRole = getLoggedInUser().userType;
+      const navigate = useNavigate();
+
+      useEffect(() => {
```
Comment (LalitKushwah) [inline src/common/Permission/RoutePermission.tsx]: Why we are using useEffect here? This RBACComponent is confusing we should make it more simple

File: src/common/Permission/CrudPermission.tsx [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+import { crudPermissions, getLoggedInUser } from '../../utils/common';
+
+const CrudPermissionwithRBAC = <P extends object>(
+  WrappedComponent: React.ComponentType<P>,
+  permission: string
+) => {
+  const WithRBAC: React.FC<P> = (props) => {
+    const userRole = getLoggedInUser().userType;
+    const [isDisabled, setIsDisabled] = useState(true);
+    useEffect(() => {
```
Comment (LalitKushwah) [inline src/common/Permission/CrudPermission.tsx]: Here also I didn't get why we are using useEffect?

File: src/services/Api.ts
Comment (lokeshkose) [reply src/services/Api.ts:1]: resolved

File: src/translations/resources.json
Comment (lokeshkose) [reply src/translations/resources.json:1]: Resolved

File: src/utils/common.ts (hunk 7 above)
Comment (lokeshkose) [reply src/utils/common.ts]: Resolved

File: src/services/Schedulers.ts (hunk 8 above)
Comment (lokeshkose) [reply src/services/Schedulers.ts]: Resolved

File: src/pages/Schedulers/Schedulers.css (hunk 9 above)
Comment (lokeshkose) [reply src/pages/Schedulers/Schedulers.css]: Resolved

File: src/common/Permission/RoutePermission.tsx (hunk 10 above)
Comment (lokeshkose) [reply src/common/Permission/RoutePermission.tsx]: Resolved

File: src/common/Permission/CrudPermission.tsx (hunk 11 above)
Comment (lokeshkose) [reply src/common/Permission/CrudPermission.tsx]: Resolved

File: src/services/Api.ts [hunk 12]
```diff
@@ -1,7 +1,7 @@
-import axios, { AxiosInstance } from 'axios';
+import axios, { AxiosError, AxiosInstance } from 'axios';
 import { message } from 'antd';
 import { NavigateFunction } from 'react-router-dom';
-const BASE_URL = process.env.REACT_APP_BASE_URL;
+const BASE_URL = 'https://order.tradekings.app:4001/dg/testing/api/';
```
Comment (LalitKushwah) [inline src/services/Api.ts]: Why we need this change?

Comment (lokeshkose) [reply src/services/Api.ts]: Fot testing on testing machine.

Comment (lokeshkose) [reply src/services/Api.ts]: Reverted

**Review by LalitKushwah — APPROVED** [review]

**Pull request merged**